message/html/compose.js

```diff
@@ -53,8 +55,17 @@ exports.create = function (api) {
       title: when(meta.channel, 'Reply is in same channel as original message')
     })
 
+    var draftPerstTimeout = null
     var textArea = h('textarea', {
-      'ev-input': () => hasContent.set(!!textArea.value),
+      'ev-input': () => {
+        hasContent.set(!!textArea.value)
+        clearTimeout(draftPerstTimeout)
+        draftPerstTimeout = setTimeout(() => {
+          // TODO: /public might not be the only place where there is no root?
+          let where = resolve(meta).root || '/public'
```

Comment:
There are a couple of cases I think you've identified well : 
- **totally new thread**: 
  1. `/public`
    - I think you're where = `/public` might be fine?
  2. `/private`
    - I think we could/ should make `meta` observeable and check for if there are `recps` loaded in. Making it observeable would mean we wouldn't be leaning on the pre-publish to add `recps` .. then we could use `where = serialize(recps, '/private')` ?  I think this might change when I finally make a nice composer for `/inbox` 
    - I'll see if I can get that working ... 
- **extending an existing thread**
  1. a straight reply - using root works fine `root`
  2. a nested reply or a fork - needs `root` and the new `fork` which I'll be implementing (maybe needs everything by `branch` from meta?

I think serialising or hashing meta would be a good `where`, because the meta describes the type of message, the root, and whether it's a nested reply in some implementations.

Buuut.. the branch might change, so maybe we want where

Reply:
with the `private` page... making meta observealbe looks like a good refactor, but not one I hav etime to do well right now.

Minimal solution ? 

pass the `location` into compose directly : 

```
  function compose ({ location, meta, prepublish, shrink = true, placeholder = 'Write a message' }, cb) { 

    var where = location || meta.root+meta.fork  // or something
```

In patchbay, routing is done with 'objects' - you'll see each page is passed a `location`, and this should be an object like `{ page: '/inbox' }`  or `{ key: '%asd1ny287xkjxw' }` or (in the future) `{ msg: '%asd1ny287xkjxw', fork: '%gga0DD2342wd }`